include/aws/s3/s3_client.h

```diff
@@ -398,12 +399,12 @@ struct aws_s3_meta_request_options {
 
     /**
      * Optional.
-     * For meta requests that support pause/resume (e.g. PutObject), the resume token returned by
+     * For meta requests that support pause/resume (e.g. PutObject), serialized resume token returned by
      * aws_s3_meta_request_pause() can be provided here.
      * Note: If PutObject request specifies a checksum algorithm, client will calculate checksums while skipping parts
      * from the buffer and compare them them to previously uploaded part checksums.
      */
-    const struct aws_byte_cursor *resume_token;
+    struct aws_s3_meta_request_resume_token *resume_token;
```

Comment:
rambling: there's sort of an argument for our _acquire()/_release() functions taking `const` pointers, even though they do modify the object's internal state, just so we can pass around const references in cases like this

but none of our acquire/release functions currently do this. Just rambling....